### PR TITLE
Add few biobb packages

### PR DESCRIPTION
**Pull request by w8jcik**
Tested by calling

```bash
spack install py-biobb-analysis py-biobb-chemistry py-biobb-gromacs
```

```python
from biobb_gromacs.gromacs.grompp import grompp
from biobb_analysis.gromacs.gmx_energy import gmx_energy
```

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-biobb-analysis/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-biobb-analysis/package.py]: Should be:
```suggestion
    depends_on("py-biobb-common@4.1.0", type=("build", "run"))
```
to get `spack install --test=root` working

File: var/spack/repos/builtin/packages/py-biobb-analysis/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-biobb-analysis/package.py]: Spack only supports Python 3.8+ anyway, so this line isn't actually required, but fine to keep it if you want.

**Review by adamjstewart — COMMENTED** [review]
Same comments for all packages

**Comment by w8jcik** [comment]
> Same comments for all packages

Corrected in all four packages.

File: var/spack/repos/builtin/packages/py-biobb-analysis/package.py
Comment (w8jcik) [reply var/spack/repos/builtin/packages/py-biobb-analysis/package.py]: Removed

File: var/spack/repos/builtin/packages/py-biobb-analysis/package.py
Comment (w8jcik) [reply var/spack/repos/builtin/packages/py-biobb-analysis/package.py]: Added

File: var/spack/repos/builtin/packages/py-biobb-common/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-biobb-common/package.py]: Should be build/run to get tests to pass

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Comment by adamjstewart** [comment]
#41993 also adds one of these packages

File: var/spack/repos/builtin/packages/py-biobb-common/package.py
Comment (w8jcik) [reply var/spack/repos/builtin/packages/py-biobb-common/package.py]: Corrected

**Comment by w8jcik** [comment]
> #41993 also adds one of these packages

We could wait until mentioned Pull Request gets merged and then I will resolve conflicts.

@d-beltran Do you want to maintain `py-biobb-{analysis,chemistry,gromacs}` as well? I am adding these packages for some past project. Not using these at the moment.

**Comment by d-beltran** [comment]
Hi @w8jcik,

Thanks for adding biobbs to spack. We may be duplicating work however.
I am working with the biobb developers to integrate biobbs in spack setting myself as maintainer. I was planning to add some more biobbs once the biobb-common is accepted, including all three additional biobbs in your pull request:
- biobb-analysis
- biobb-common
- biobb-chemistry
- biobb-gromacs
- biobb-io
- biobb-model
- biobb-structure-checking
- biobb-structure-utils

Besides I was planning to add 2 more packages for dependencies:
- acpype
- ambertools

And 2 last packages used widely in biobb workflows for demonstration:
- nglview
- simpletraj

I have these spack packages already implemented and tested in the [EBRAINS collab](https://www.ebrains.eu/tools/collaboratory).

I see you did not include dependencies beyond biobb-common and setuptools in your packages. Is this intended?
In my implementation I added these dependencies:
- biobb-gromacs: gromacs (already in spack) 
- biobb-chemistry: openbabel (already in spack), ambertools and acpype
- biobb-analysis: gromacs and ambertools

Best regards

**Comment by w8jcik** [comment]
@d-beltran 

> I see you did not include dependencies 

In my workflow I was using manually modified and installed GROMACS and I completely forgot that I should ship GROMACS as a dependency. Similar with `acpype`.

> I have these spack packages already implemented and tested

In such case it is the best to just close this merge request.

I am looking forward to use the new biobb packages :)

Note: There is also a pull request for `py-nglview` https://github.com/spack/spack/pull/42079. Not sure if it also a duplicate. It is slightly older version due to missing dependencies. It is also missing possible variants. Should I close that as well?

**Comment by adamjstewart** [comment]
Note that it's not necessary to submit separate PRs for each package. If package A depends on package B, you can add both A and B in a single PR.

**Comment by d-beltran** [comment]
> Note: There is also a pull request for py-nglview https://github.com/spack/spack/pull/42079. Not sure if it also a duplicate. It is slightly older version due to missing dependencies. It is also missing possible variants. Should I close that as well?

Not at all. My package version is even older (3.0.4) and dependencies are very similar so it would be perfect if you could finish your PR. I'll be the first user :)

**Pull request closed without merging**
